src/main/java/com/squareup/javapoet/CodeWriter.java

```diff
@@ -254,6 +290,29 @@ public CodeWriter emit(CodeBlock codeBlock) throws IOException {
     return this;
   }
 
+  private static String gleanMemberName(String part) {
+    for (int i = 1; i <= part.length(); i++) {
+      if (!SourceVersion.isIdentifier(part.substring(0, i))) {
+        return part.substring(0, i - 1);
+      }
+    }
+    return part;
+  }
+
+  private boolean emitStaticImportMember(String canonical, String part) throws IOException {
+    String partWithoutLeadingDot = part.substring(1);
+    char first = partWithoutLeadingDot.charAt(0);
```

Comment:
Dooh. Late time optimization, the `.isEmpty()` check is too late now. Gonna fix it with a test case attached.